doc/specs/#605 - Search/spec.md

```diff
@@ -0,0 +1,109 @@
+
+## UI/UX Design
+
+![SearchBox mockup](images/SearchBoxControl.png)
+
+Above is the SearchBoxControl. The two buttons with up/down arrows controls the search direction, it will be styled once clicked. The checkbox, if checked, means that we are searching case-sensitivily. The Dockbottom button on the right of the checkbox is for moving the search box to top/bottom. 
+
+![SearchBox mockup, arrow button clicked](images/SearchBoxUpSelected.png)
+
+The search box defaults to be on the top right corner of the Terminal window. If the current tab is split into panes, each pane will have a individual searchbox. The user can move it to bottom to avoid it blocking too much screen contents. 
+
+## Capabilities
+
+1. The user can search exact matched text in the text buffer of the Terminal Screen. 
+2. The user can choose to search case sensitively and insensitively. 
+3. The user can search up or down. 
+4. Found text will be selected. 
+5. The search will start from the active selected text (inclusive) if there is one, or where the cursor is. 
+5. The search will automatically go around when it reaches the starting point.
+7. The search box can be positioned at either the top left or the bottom left of the terminal. 
+8. The user can use Tab to navigate all the elements in the search box. 
+
+### Accessibility
+
+The user should be able to use search by keyboard only.
+Once the searchbox is focused, the user can navigate between elements in the search box using Tab. And "click" using Enter. 
+
+### Security
+
+This feature should not introduce any new security issues.
+
+### Reliability
+
+1. The key input of Terminal command line and the search box should be separated. search box should not block interaction with the command line when it is open. 
+2. The search box should not block text. This issue could be addressed by make search box movable. 
+
+### Compatibility
+
+This feature won't break existing features of Terminal.
+
+### Performance, Power, and Efficiency
+
+This feature only launches in need. It does not impact the performance of Terminal. 
+
+## Potential Issues
+
+1. The search bar should not block command line view. 
+2. Search should not block any program the Terminal is currently executing.
+
+## Future considerations
+
+In version 1, we want realize a case sensitive/insensitive exact text match. But we may consider the following features in version 2:
+
+1. Add "Find" button in dropdown menu to trigger search. 
+2. Search from all tabs. For Version 1 we just want to realize search within one tab. However, the community also requests search from all tabs. We put in our goals for Version 2. 
+3. Regular experssion match. 
+4. Search history.
+5. High-light while you type. Emphasizing all the other matches in the buffer with an outline or selection with another color. 

```

Comment:
Add some more context here. Why would we want this feature? Why is it not being considered part of this spec currently? (this comment really applies to each of the bullet points in this list)

Reply:
Added